src/bitcoin/hash.h

```diff
@@ -170,10 +170,26 @@ class CHashWriter {
         ctx.Write((const uint8_t *)pch, size);
     }
 
-    // invalidates the object
+    /** Compute the double-SHA256 hash of all data written to this object.
+     *
+     * Invalidates this object.
+     */
     uint256 GetHash() {
         uint256 result;
-        ctx.Finalize((uint8_t *)&result);
+        ctx.Finalize(result.begin());
+        ctx.Reset()
+            .Write(result.begin(), CSHA256::OUTPUT_SIZE)
+            .Finalize(result.begin());
+        return result;
+    }
+
+    /** Compute the SHA256 hash of all data written to this object.
+     *
+     * Invalidates this object.
+     */
+    uint256 GetSHA256() {
+        uint256 result;
```

Comment:
This codebase has a `uint256::Uninitialized` c'tor tag you can use.  This reduces overhead by not setting the `uin256` to all zero's before being written-to again by `Finalize()` (which is guaranteed to fill it with bytes)....

What you have here potentially writes to the same memory twice -- although in this codebase the default `uint256()` c'tor is `constexpr` so it's possible that the compiler may elide the initial zeroing construction completely...